src/devices_models/devices/AC_branches.jl

```diff
@@ -109,52 +110,66 @@ function branch_rate_constraints!(
     ::Nothing,
-) where {B <: PSY.ACBranch}
-    range_data = [(PSY.get_name(h), PSY.get_rate(h)) for h in devices]
-    rating_constraint!(
-        container,
-        range_data,
-        RateLimitFTConstraint(),
-        (FlowActivePowerFromToVariable(), FlowReactivePowerFromToVariable()),
-        B,
-    )
+) where {B <: PSY.ACBranch, T <: PM.AbstractPowerModel}
+    rating_data = [(PSY.get_name(h), PSY.get_rate(h)) for h in devices]
 
-    rating_constraint!(
-        container,
-        range_data,
-        RateLimitTFConstraint(),
-        (FlowActivePowerToFromVariable(), FlowReactivePowerToFromVariable()),
-        B,
-    )
-    return
+    time_steps = get_time_steps(container)
+    var1 = get_variable(container, FlowActivePowerFromToVariable(), B)
+    var2 = get_variable(container, FlowReactivePowerFromToVariable(), B)
+    add_cons_container!(container, cons_type(), B, [r[1] for r in rating_data], time_steps)
+    constraint = get_constraint(container, cons_type(), B)
+
+    for r in rating_data
+        for t in time_steps
+            constraint[r[1], t] = JuMP.@constraint(
+                container.JuMPmodel,
+                var1[r[1], t]^2 + var2[r[1], t]^2 <= r[2]^2
+            )
+        end
+    end
 end
 
-function _branch_flow_constraint!(
-    jump_model::JuMP.Model,
-    ptdf_col::Vector{Float64},
-    nodal_balance_expressions,
-    flow_variables,
-    t::Int,
-)
-    return JuMP.@constraint(
-        jump_model,
-        sum(ptdf_col[i] * nodal_balance_expressions[i, t] for i in 1:length(ptdf_col)) -
-        flow_variables[t] == 0.0
-    )
+function add_constraints!(
+    container::OptimizationContainer,
+    cons_type::Type{RateLimitTFConstraint},
+    devices::IS.FlattenIteratorWrapper{B},
+    model::DeviceModel{B, <:AbstractBranchFormulation},
+    ::Type{T},
+    ::Nothing,
+) where {B <: PSY.ACBranch, T <: PM.AbstractPowerModel}
+    rating_data = [(PSY.get_name(h), PSY.get_rate(h)) for h in devices]
+
+    time_steps = get_time_steps(container)
+    var1 = get_variable(container, FlowActivePowerToFromVariable(), B)
+    var2 = get_variable(container, FlowReactivePowerToFromVariable(), B)
+    add_cons_container!(container, cons_type(), B, [r[1] for r in rating_data], time_steps)
+    constraint = get_constraint(container, cons_type(), B)
+
+    for r in rating_data
+        for t in time_steps
+            constraint[r[1], t] = JuMP.@constraint(
+                container.JuMPmodel,
+                var1[r[1], t]^2 + var2[r[1], t]^2 <= r[2]^2
+            )
+        end
+    end
 end
 
-function branch_flow_values!(
+function add_constraints!(
     container::OptimizationContainer,
+    cons_type::Type{NetworkFlowConstraint},
     devices::IS.FlattenIteratorWrapper{B},
-    ::DeviceModel{B, <:AbstractBranchFormulation},
-    network_model::NetworkModel{StandardPTDFModel},
-) where {B <: PSY.ACBranch}
+    model::DeviceModel{B, <:AbstractBranchFormulation},
+    network_model::NetworkModel{S},
```

Comment:
note that this signature differs from the other `add_constraints!` methods because we need the entire `network_model` to get the PTDF

Reply:
good stuff here with the PTDF.